protos/camera_server/camera_server.proto

```diff
@@ -58,6 +85,55 @@ message StopVideoResponse {
     int32 stream_id = 1;    // video stream id
 }
 
+message SubscribeStartVideoStreamingRequest {}
+message StartVideoStreamingResponse {
+    int32 stream_id = 1;    // video stream id
+}
+
+message SubscribeStopVideoStreamingRequest {}
+message StopVideoStreamingResponse {
+    int32 stream_id = 1;    // video stream id
+}
+
+message SubscribeSetModeRequest {}
+message SetModeResponse {
+    Mode mode = 1;
+}
+
+message SubscribeStorageInformationRequest {}
+message StorageInformationResponse {
+    int32 storage_id = 1;
+}
+
+message RespondStorageInformationRequest {
+    StorageInformation storage_information = 1; // camera storage information
+}
+message RespondStorageInformationResponse {
+    CameraServerResult camera_server_result = 1;
+}
```

Comment:
And then these shouldn't be required because the response is sent in the response to the subscription.

Or do I have that completely backwards in my head now @JonasVautherin?

Reply:
I have some misunderstood about the camera server design.

For now I add all respond command with feedback.

Reply:
But that's different from how it is done in other server plugins, no?

Reply:
Oh, I just use the origin take photo way to implment other camera function. Can you provide another suggestion way to implment the interface.

Reply:
Like this for instance?

https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/action_server/action_server.proto#L74C9-L77

https://github.com/mavlink/MAVSDK/blob/main/src/mavsdk/plugins/action_server/action_server_impl.cpp#L63-L96

Reply:
Does that make sense @tbago ? Do you understand what I mean?

Reply:
Hi julian, I try to undersand the action server implement. Sorry for my poor english, I just draw a graph to show what I mean.
``` mermaid
---
title: Action Process
---
flowchart LR
    subgraph MAVSDK Client
    C1[Action]
    end
    
    subgraph MavSDK Server
    M1[Receive Action message]
    M2[Process action]
    M1<-->M2
    end

    subgraph ActionSubscribe
    B1[subscribe result]
    end

    C1 <-->|mavlink| M1
    M2 <--> |subscribe| B1
```
The [subscribe arm](https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/action_server/action_server.proto#L13) is just callback function for subscribe the action server result and arm state. And the action result is process by [this code](https://github.com/mavlink/MAVSDK/blob/main/src/mavsdk/plugins/action_server/action_server_impl.cpp#L70C23-L90).

Reply:
For the camera function. I just want to use the [camera server](https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/camera_server/camera_server.proto) like middleware. And the camera process logic will be in the backend, like the [example code](https://github.com/tbago/MAVSDK/blob/add_more_camera_function/examples/camera_server/camera_server.cpp#L111). And I will use [respond message](https://github.com/tbago/MAVSDK-Proto/blob/add_more_camera_function/protos/camera_server/camera_server.proto#L27) to tell the **MAVSDK Camera Server** process result.

Reply:
Do you understand what I mean ? And maybe I have some missundend about the action server.

Reply:
Your English is fine, don't worry! The part I don't understand is why you want to implement it differently for the camera server, or why you can't use the same way it is done in the ActionServer? I would like to have the servers consistent/all the same, so that it is easy to implement against.

Reply:
Yeah, I think the same way to implement the server is good disign. But for our camera usage, all the camera function and response result is platform dependency, So we can just use the CameraServer as a delegate.
For other camera implement may have the same design, the camera is all running on specail embeded chip and have different sdk.

Reply:
Hmm, I don't really want to make an exception because of some special embedded camera SDK. What's the problem, restriction that you run into? Surely, this is solvable with software? Or can you make a code example to illustrate the problem?

Reply:
Hi Julianoes, sorry for long time no replay:
I just draw another sequence diagram to show the basic logic (this logic is already implement in the [MAVSDK](https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/camera_server/camera_server.proto#L18-L21)) 
``` mermaid
sequenceDiagram
    participant c as MAVClient
    participant s as MAVServer
    participant b as Backend(running on embeded chip)
    loop
        b-->>s: subscribe take photo callback
        c-->>s: call take photo
        s-->>b: callback(the pre subscribe method)
        
        Note right of b: process the take photo command
        Note right of b: call embeded chip sdk to really take photo

        alt respond take photo 
            b->>s: Success take photo
        else Failed
            b->>s: Interal embeded system error
        else Busy
            b->>s: Not ready take photo
        end

        s->>c: responed take photo result
    end
  
```

Reply:
On more question:
First I am not familiar with the action server, so maybe I have some misunderstand. For example if I have a real drone (running with the action server), and I want to arm and take off the drone, so first I need check the drone state(like battery is ok).
So the action server also need to communicate the drone (running on embeded system) to get state.

Reply:
@JonasVautherin do you have an opinion here? Maybe I'm confused.

Reply:
I think I was confused, and @tbago was right.